### DISCUSSION TITLE
Enterprise Wi-Fi Support

### DISCUSSION THREAD

**Discussion by mlhyyl** [arendst/Tasmota#13372] [Ideas]
I'm trying to set up my Tasmota at my workplace. The problem is, after selecting the Wifi network, only one field for a password is provided. However, our enterprise network requires a username and password to be provided. I've tried to hardcode it with platform.io but couldn't find a field that would provide input for the username.

Is there an alternative way or is this a dead-end?

**Comment by barbudor** [comment]
Enterprise authentication ([EAP](https://en.wikipedia.org/wiki/Extensible_Authentication_Protocol)) are not supported by Tasmota.
Not sure what level of support is provided in the Arduino Core.

Which exact EAP method does your network requires ?

**Comment by mlhyyl** [comment]
@barbudor I get the network listed as a WPA&WPA2 Enterprise with Tunneled TLS as authentication and MSCHAPv2 (no EAP) as inner authentication. Hope that's what you were asking for.

**Comment by barbudor** [comment]
Thanks.
It seems that it is supported in the Espressif SDK since some time and Arduino Core also.
I don't know what impact on code size it would have to support it. Probably not in mainstream builds

@Jason2866 Any opinion on that ?
This thread seems interresting https://gist.github.com/Matheus-Garbelini/2cd780aed2eddbe17eb4adb5eca42bd6
But I have no access to such network to even test the base code

**Comment by Jason2866** [comment]
@barbudor you already told what i know. Supported from IDF and Arduino. @ascillato did the changes in wifi part for Tasmota. So he and Theo has the best knowledge.
Setting up a Enterprise Wlan is not done in a few minutes...

**Comment by ascillato** [comment]
@mlhyyl

There is no plan to add Enterprise authentication at least in the next months. If you want to provide a PR you are very welcome.
